File: SantanderTeste/app/src/main/java/com/example/jcsantos/santanderteste/Modules/Login/LoginService.java
package com.example.jcsantos.santanderteste.Modules.Login;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.Headers;
import retrofit2.http.POST;

public interface LoginService {
    @Headers({"Content-Type: application/x-www-form-urlencoded"})
    @FormUrlEncoded
    @POST("login")
    Call<LoginModel> loginService(
            @Field("user") String user,
            @Field("password") String password
    );

}

File: app/src/main/java/pt/felipegouveia/bankapp/data/login/api/LoginService.kt
package pt.felipegouveia.bankapp.data.login.api

import io.reactivex.Single
import pt.felipegouveia.bankapp.data.login.model.LoginBody
import pt.felipegouveia.bankapp.data.login.model.LoginResponse
import retrofit2.http.*

/**
 * LoginService - Provides methods to interact with login endpoints of BankApp API
 */
interface LoginService {

    /**
     * Method to login using @see LoginBody
     *
     * @return RxJava `Single` for the login response.
     */
    @Headers("Content-Type: application/x-www-form-urlencoded")
    @POST("login")
    fun login(
        @Body
        loginBody: LoginBody
    ): Single<LoginResponse>
}
